enterprise/signals/signal_base.py

```diff
@@ -1188,12 +1190,12 @@ def _get_logdet(self):
             logdet = np.sum(np.log(self._nvec[self._idx]))
         else:
             logdet = 0
-        for slc, block in zip(self._slices, self._blocks):
-            if slc.stop - slc.start > 1:
-                cf = sl.cho_factor(block + np.diag(self._nvec[slc]))
+        for idx, block in zip(self._idxs, self._blocks):
+            if len(idx) > 1:
+                cf = sl.cho_factor(block + np.diag(self._nvec[idx]))
                 logdet += np.sum(2 * np.log(np.diag(cf[0])))
             else:
-                logdet += np.sum(np.log(self._nvec[slc]))
+                logdet += np.sum(np.log(self._nvec[idx]))
```

Comment:
This line is missing in the tests

Reply:
As you can see, the line it replaces was also missing in the test. This is just a replacement of slc with idx.

In order for us to cover this line, we need to add a NANOGrav test set that has only a single observation in an epoch.

My vote would be to leave the test coverage as it is, because IMHO this if statement should disappear. Even if len(idx)==1 we should include the ecorr. See issue #355